cloudwash/providers/aws.py

```diff
@@ -53,6 +53,27 @@ def dry_discs():
                     [dry_data["DISCS"]["delete"].append(ddisc["VolumeId"]) for ddisc in rdiscs]
                 return rdiscs
 
+            def dry_images():
+                rimages = []
+                if settings.aws.criteria.image.unassigned:
+                    rimages = aws_client.list_templates(executable_by_me=False, owned_by_me=True)
+                    free_images = aws_client.filter_free_images(
+                        image_list=[image.raw.image_id for image in rimages]
+                    )
+                    remove_images = [
+                        image
+                        for image in free_images
+                        if image not in settings.aws.exceptions.images
+                    ]
+                    if settings.aws.criteria.image.delete_pattern:
+                        remove_images = [
+                            image
+                            for image in remove_images
+                            if not image.startswith(settings.aws.criteria.image.delete_pattern)
```

Comment:
This `not` looks backwards.

If I have a list of free images that aren't in the exception list, and a `delete_pattern` from config that dictates what image names to delete, this gives me a list of names to keep, not names to remove.

```
# list of free images not excluded
In [1]: free_list = ['deleteme-one', 'image-two', 'image-three', 'deleteme-four']

# assuming delete_pattern is 'deleteme'
In [2]: remove_images = [i for i in free_list if not i.startswith('deleteme')]

# I end up with the images that should keep, not that should be deleted?
In [3]: remove_images
Out[3]: ['image-two', 'image-three']

```

Reply:
Maybe this comes from an intent to have the default behavior pick all images for removal if the `delete_pattern` is empty?

Either way, it needs to be adjusted so `delete_pattern` is applied correctly when present - I would recommend actually using `re` here and actually accept a pattern string. Using just `startswith` will immediately limit the use and dictate that any teams wanting to use this tool have to prefix image names.

`re.compile` can be used to create the regex pattern object (and to parse the pattern from config), and then you can use  `Pattern.search` function in the list comprehension to indicate whether the given pattern is anywhere in the string. `match` can be used too, but will be more explicit and will require the caller to specify `.*` in their pattern to match anywhere in the string.  I think either function is fine here, but the docblock/readme/config comment should explain which is in use.

Reply:
Also agree with Mike here, with `not` we are doing opposite here! Please remove `not` from the condition!

Reply:
Removed not.
As for delete_pattern, I am moving away from it and keep it what cloudwash already has. Deleting images starting with a keyword mentioned in conf.